### ISSUE TITLE
Regarding the issue of data format conversion.

### ISSUE THREAD

**Issue by lienji** [FongMunHong/FlowER#4]
Hello,

I noticed that the test data provided on GitHub is in the following SMILES format:  
`O.Cl.[OH3+].CC(=O)CC(=O)C(F)(F)F.NNc1cccc(Br)c1>>Cc1cc(n(n1)-c1cccc(c1)Br)C(F)(F)F|Cc1cc(nn1-c1cc(ccc1)Br)C(F)(F)F`.

However, my `test.txt` file uses atom-mapped SMILES, such as `[C:19]([O:20][H:36])...`.

I would like to evaluate my model on `test.txt`, but after converting the atom-mapped format to the standard reaction SMILES (like the GitHub example), my test accuracy drops to only 5%.

Could you please provide a reliable format conversion script that properly transforms atom-mapped reaction SMILES (e.g., `[C:1]...>>[C:1]...`) into the clean, unmapped format used in the official test set?

Thank you very much!

**Comment by FongMunHong** [comment]
Hi, 

Thanks for your interest in our work.

I am not sure if I quite follow your question. Are you trying to use beam search or are you trying to do inference?
- Beam search process would not use mapping, as you would give it the initial reactant pool and it will perform propagate till it -reaches the specified product.
- Inference process requires you to specify all intermediates with mapping, to check if the model recovers all intermediates. 

All of the atom mapping is done using RxnMapper, which is proprietary software (that we unfortunately could not share) but is available for purchase. 

Best,
Mun Hong

**Comment by lienji** [comment]
I'm currently running the script FlowER-master/examples/run_beam_predict_seed.sh, which expects input in the format used by FlowER-master/data/flower_new_dataset/beam.txt, e.g.:

O.Cl.[OH3+].CC(=O)CC(=O)C(F)(F)F.NNc1cccc(Br)c1>>Cc1c...,
However, the format of FlowER-master/data/flower_new_dataset/test.txt (and the training data) is different—it uses atom-mapped SMILES like:

[C:19]([O:20][H:36])([C:21]([H:37])([H:38])...
I’ve modified the code to directly accept test.txt-style input, but my success rate for product prediction is only around 50% (where I define "success" as generating all correct products exactly).

Could you please share:

A script to convert the test.txt format  into the beam.txt format ?
Your exact evaluation criteria or success definition used during testing?
This would help me better align my evaluation with yours and identify where the issue lies.

**Comment by FongMunHong** [comment]
Hi,

`test.txt` is for inference of the model, not _exactly_ used for beam search.  It is used to reproduce our result in our paper `Figure 2b & c`.  Also, `test.txt` comprises of all the intermediates in the reaction sequence, which is not best used for beam search since it **expects** only the first reactant. 

You are right to observe 50% of the time success, since this generation process is a _stochastic_ process. Beam search serves as idea generator of what the possible intermediates would be and what kind of products it would end up with. This is specifically used in Figure 3b in our paper. 

Note, which is possibly your main point of confusion, that we are **NOT** using `run_beam_predict_seed.sh` to get `Figure 2b & c`, (if that's what you mean by "success" criteria), and we are using `eval.sh` to evaluate our model on recovering every exact intermediate. Beam search is only use to generate potential pathways which in our use case to check and see if it generalizes to unseen reaction class in training, which we show only **one** example through Figure 3b.

As for converting mapped rxn_smiles to non mapped rxn_smiles, you can use this.  Remember to split using '>>'. 
https://github.com/FongMunHong/FlowER/blob/72d5b407c39aebc6ad9926d09d6e85133c200ba7/sequence_evaluation.py#L33-L38

Take note if you want to use `test.txt` to do beam search for evaluation (which our paper **never** did), you have to first group `text.txt` based on their id (each test.txt line is in format `rxn_smiles|id`),  then build a directed acyclic graph, and identify which is the initial pool of reactant, and the final pool of products. You can use `sequence_evaluation.py` to help you with constructing the pipeline.

**Comment by lienji** [comment]
Thank you very much for your response—it is extremely helpful for my work.
